app/controllers/products_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ProductsController < ApplicationController
+  def new
+    @product = Product.new
+    @vendor = Vendor.find(params[:vendor_id])
+  end
+
+  def create
+    product = Product.create(product_params) do |p|
```

Comment:
Good use of a block to perform additional configuration for the new `Product` instance.